lib/components/forms/GKECredentialsForm.js

```diff
@@ -48,6 +51,31 @@ class GKECredentialsForm extends React.Component {
     this.setState(state)
   }
 
+  async verify(name, tryCount) {
+    tryCount = tryCount || 0
+    if (tryCount === 0) {
+      message.loading({ content: 'Verifying credentials', key: 'verify', duration: 0 })
+    }
+    if (tryCount === 3) {
+      message.error({ content: 'Verification failed', key: 'verify' })
+      const state = copy(this.state)
+      state.inlineVerificationFailed = true
+      state.submitting = false
+      state.formErrorMessage = 'These credentials could not be verified, please try again or save anyway.'
+      this.setState(state)
+    } else {
+      setTimeout(async () => {
+        const gkeResult = await apiRequest(null, 'get', `/teams/${this.props.team}/gkecredentials/${name}`)
+        if (gkeResult.status.status === 'Success') {
+          message.success({ content: 'Verification successful', key: 'verify' })
+          return await this.props.handleSubmit(gkeResult)
+        } else {
+          return await this.verify(name, tryCount + 1)
+        }
+      }, 2000)
+    }
+  }
+
```

Comment:
Not sure about this assumption that if it fails three times with 2 second interval being very solid? would it be worth only retrying if the status is still verifying/pending, but retrying for longer if it is still pending/verifying?

Reply:
After our slack discussion yesterday and speaking to Rohith this morning I think this is safe enough for now. It can be tweaked in the future if we find it's not enough time.

If for some reason the verification cannot be done eg the GCP API is down, it will show the user the error and allow them to continue anyway.

Reply:
okies, will approve based on that.

Reply:
Thanks, I added a screenshot to show what the error looks like